app/src/main/java/org/oppia/android/app/devoptions/ForceDownloadRemoteParametersDialogFragmentPresenter.kt

```diff
@@ -0,0 +1,78 @@
+package org.oppia.android.app.devoptions
+
+import android.app.AlertDialog
+import android.app.Dialog
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import org.oppia.android.app.databinding.databinding.ForceDownloadRemoteParametersDialogFragmentBinding
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import org.oppia.android.app.ui.R
+import org.oppia.android.domain.devoptions.ForceDownloadRemoteParametersController
+import org.oppia.android.domain.oppialogger.OppiaLogger
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import javax.inject.Inject
+
+/** Presenter for the [ForceDownloadRemoteParametersDialogFragment]. */
+class ForceDownloadRemoteParametersDialogFragmentPresenter @Inject constructor(
+  private val fragment: Fragment,
+  private val activity: AppCompatActivity,
+  private val oppiaLogger: OppiaLogger,
+  private val resourceHandler: AppLanguageResourceHandler,
+  private val forceDownloadRemoteParametersParametersController:
+    ForceDownloadRemoteParametersController
+) {
+  /**
+   *  Creates a dialog to display the status of the ongoing remote parameters download and then
+   *  restart.
+   */
+  fun handleOnCreateDialog(): Dialog {
+    val binding = ForceDownloadRemoteParametersDialogFragmentBinding.inflate(
+      activity.layoutInflater,
+      /* parent= */ null,
+      /* attachToRoot= */ false
+    )
+    binding.lifecycleOwner = fragment
+    binding.isRestartEnabled = false
+
+    forceDownloadRemoteParametersParametersController
+      .downloadRemoteParameters().toLiveData().observe(fragment) {
+        when (it) {
+          is AsyncResult.Success -> {
+            oppiaLogger.d(
+              "ForceDownloadRemoteParametersDialog",
+              "Remote parameters downloaded successfully."
+            )
+            handleDownloadComplete(binding)
+          }
+          is AsyncResult.Failure -> {}
+          is AsyncResult.Pending -> {}
+        }
+      }
+
+    val dialog = AlertDialog.Builder(activity, R.style.OppiaAlertDialogTheme)
+      .setView(binding.root)
+      .create()
+    dialog.setCanceledOnTouchOutside(false)
+
+    binding.restartButton.setOnClickListener {
+      (activity as DeveloperOptionsActivity).developerOptionsActivityPresenter.markRestartRequired()
```

Comment:
This should change when you move the function to this class.
```suggestion
    markRestartRequired()
```

Reply:
need clarification on this, PTAL at this [comment](https://github.com/oppia/oppia-android/pull/5949/#discussion_r2328230964)